azurelinuxagent/ga/exthandlers.py

```diff
@@ -947,6 +973,17 @@ def collect_ext_status(self, ext):
 
         return ext_status
     
+    def is_ext_status_success(self, ext):
+        status = self.collect_ext_status(ext)
+        if status is None:
+            return (True, status)
```

Comment:
Seems odd, why is this `True`?

Reply:
Some extensions do not report any status. We should not wait for such extensions to report status. That's why we return True here.

Reply:
But by returning True here we are saying that a extension that has not reported status yet has succeeded... correct?

Reply:
Yes. I think we can return False in this case. @uiri do you have any comment on this?

Reply:
`collect_ext_status` returns None if the extension does not report any status. Not all extensions report status.

If you return `False` in this case, the agent will hang until it times out on such extensions.

Reply:
Base on offline discussion with @uiri , an extension can install and enable but may not create any status file. In this case, we don't want to wait for it until it is timed out. So we return True in this case.

Reply:
TBD: What should we return if there is no status file?